internals/overlord/logstate/gatherer.go

```diff
@@ -383,9 +383,13 @@ func newLogClient(target *plan.LogTarget) (logClient, error) {
 			ScopeName:  cmd.ProgramName,
 		}), nil
 	case plan.SyslogTarget:
+		hostname, err := os.Hostname()
+		if err != nil {
+			return nil, fmt.Errorf("cannot get hostname: %w", err)
+		}
 		return syslog.NewClient(&syslog.ClientOptions{
-			TargetName: target.Name,
-			Location:   target.Location,
+			Location: target.Location,
+			Hostname: hostname,
```

Comment:
Per my [earlier comment](https://github.com/canonical/pebble/pull/725#discussion_r2566684674), we should pass in `cmd.ProgramName` here:

```go
SDID: cmd.ProgramName,
```